### PR TITLE
Setup a Stackdriver metric adapter

### LINKED ISSUE #41: Add metrics
Different developers/hosts may want different libraries/platforms for metrics

We should probably make this an interface and include a GCP implementation to start

### PR DESCRIPTION
**Pull request by bwillard**
Fixes #41.

### CODE REVIEW

File: extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricExtension.java [hunk 1]
```diff
@@ -19,12 +19,18 @@
...
+
 public class GoogleDtpInternalMetricExtension implements ServiceExtension {
 
   @Override
   public void initialize(ExtensionContext context) {
+    try {
     context.registerService(
         DtpInternalMetricRecorder.class,
-        new GoogleDtpInternalMetricRecorder(context.getMonitor()));
+        GoogleDtpInternalMetricRecorder.getInstance());
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricExtension.java]: Is the static initialization required for a particular scenario or can we obtain dependencies and instantiate via a public constructor?

File: extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricRecorder.java [hunk 2]
```diff
@@ -15,28 +15,206 @@
...
+  }
+
+  private GoogleDtpInternalMetricRecorder() throws IOException {
+    // Enable OpenCensus exporters to export metrics to Stackdriver Monitoring.
+    // Exporters use Application Default Credentials to authenticate.
+    // See https://developers.google.com/identity/protocols/application-default-credentials
+    // for more details.
+    StackdriverStatsConfiguration configuration = StackdriverStatsConfiguration.builder()
+        .setCredentials(GoogleCredentials.getApplicationDefault())
+        .setProjectId(GoogleCloudExtensionModule.getProjectId())
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricRecorder.java]: Can we pass in both of these as constructor args?

**Comment by bwillard** [comment]
PTAL?

File: extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricRecorder.java [hunk 3]
```diff
@@ -15,28 +15,211 @@
...
+      "ms");
+
+  private static final Measure.MeasureLong GENERIC_BOOLEAN = Measure.MeasureLong.create(
+      "generic_boolean",
+      "A generic counter that services can use to hold arbitrary boolean metrics",
+      "count");
+
+  private final ViewManager viewManager;
+
+  // This is needed because Stackdriver can only be initialized once and the
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricRecorder.java]: Ok, we should work through the Google cloud packages initialization story at some point but this is very cool for now.

**Pull request merged**